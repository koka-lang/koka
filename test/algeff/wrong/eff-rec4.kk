 
effect state {
  fun get() : (int -> state int)
  fun set(x : int -> state int) : ()
}

val new-ref = handler(s) {
  get() -> resume(s,s)
  set(x) -> resume((),x)
}

fun recurse() {
  using new-ref(id)
  fun foo(i : int) {
    get()(i)
  }
  set(foo)
  foo(42)
}
