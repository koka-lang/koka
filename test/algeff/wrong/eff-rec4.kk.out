test/algeff/wrong/eff-rec4.kk(2, 1): error: Type .ops-state is declared as being (co)inductive but it occurs
 recursively in a negative position.
 hint: declare it as a 'type rec' (or 'effect rec)' to allow negative occurrences